### DISCUSSION TITLE
defaultSearchPlatform does not work

### DISCUSSION THREAD

**Discussion by Shhoyyo** [MenuDocs/erela.js#198] [Q&A]
Setting defaultSearchPlatform as Youtube music or Soundcloud  does not work as it seems as if the search platform is permeately set to youtube. Any fix for this?

**Comment by canarado** [comment]
The search function can take a SearchQuery Object which has a `source` property, this is what you want. i.e.

```js
manager.search({ query, source: "soundcloud" }, requester)
```

I just started using this library today so I am not sure about Youtube music, but the typings say Youtube and SoundCloud are supported.

**Comment by N1C0exe** [comment]
~~In order to use sc as default search plat u need to  go to erela.js / dist / structures / Manager.js and delete     youtube: "yt" in : 
 const sources = {
                soundcloud: "sc",
                youtube: "yt",
                         };        
Also in the line above u need to change "youtube" --> "souncloud " ( in the const source ) . Now every song u search the bot will search it on sc .~~

Update : Sc added a premium feature called GO+ so u need to pay in order to listen full song ;(

**Comment by viztea** [comment]
The `defaultSearchPlatform` option works completely fine for me... you're probably using a custom source when calling Manager#search
